packages/medusa-react/src/hooks/admin/products/mutations.ts

```diff
@@ -96,7 +97,7 @@ export const useAdminUpdateVariant = (
   options?: UseMutationOptions<
     Response<AdminProductsRes>,
     Error,
-    AdminPostProductsProductVariantsReq & { variant_id: string }
+    AdminPostProductsProductVariantsVariantReq & { variant_id: string }
```

Comment:
suggestion: shouldn't the variant_id be added as a parameter of the method like we did in all the other hooks? wdyt? It creates quite a lot of disturbance to have it here

Reply:
@adrien2p You mean like this `useAdminUpdateVariant("prod_something", "variant_something")? I mean we definitely could, but although a minor change, it would still be a breaking change in `medusa-react`. So I am not really sure if it is appropriate atm. What do you think @olivermrbl?

Reply:
You can go for the usual approach with the deprecated API and the new API in order to be able to remove it one day instead of being stuck 😂 wdyt. something along the line of 
```ts
/**
 *
 * @param productId
 * @param optionsOrVariantId - Deprecated: will become variantId in the next major release
 * @param options
 */
export const useAdminUpdateVariant = (
  productId: string,
  optionsOrVariantId?: string | UseMutationOptions<
    Response<AdminProductsRes>,
    Error,
    AdminPostProductsProductVariantsVariantReq & { variant_id: string }
  >,
  options?: UseMutationOptions<
    Response<AdminProductsRes>,
    Error,
    AdminPostProductsProductVariantsVariantReq & { variant_id: string }
  >
) => {
  const variantId = typeof optionsOrVariantId === "string" ? optionsOrVariantId : undefined
  options = typeof optionsOrVariantId === "string" ? options : optionsOrVariantId

  const { client } = useMedusa()
  const queryClient = useQueryClient()

  return useMutation(
    ({
      variant_id,
      ...payload
    }: AdminPostProductsProductVariantsVariantReq & { variant_id: string }) =>
      client.admin.products.updateVariant(productId, variant_id ?? variantId, payload),
    buildOptions(
      queryClient,
      [adminProductKeys.lists(), adminProductKeys.detail(productId)],
      typeof options
    )
  )
}
```

Reply:
I'm okay with adding to it, just playing the devil's advocate here:

If the proposed change is only for clarity, I think the suggested change is a bit more "disturbing" for the end user. 
If the user does not pay attention to the `deprecated` message then they might be a bit lost, as we are throwing away any meaningful IntelliSense by adding this, typescript will now be perfectly happy with accepting `("prod_id", { ...options }, { ...options })`. Since we don't have any documentation for using the different `medusa-react` hooks, I think types that are able to tell you if you are using it wrong, are the better DX. So in that regard, I think it would be better to just throw this into the backlog and update it when plan to do a major version of medusa-react

Reply:
the other solution would be to create a function overload, which would resolve both of your point 👍

Reply:
I am ok of not it in that PR but at least adding a toto + linear issue. The only problem with doing it right before a major release is that the consumer does not have time to prepare and are forced to do the changes in order to upgrade. While with the deprecated approach, they can move slowly to the new API.

Reply:
Let me know if I succeed to change your mind 😇

Reply:
I'm fine with both approaches, but introducing breaking changes to solve a typing issue is not ideal, so I'd rather save it for a separate PR and focus on getting this one included in the upcoming release 😊

Reply:
linking the pr here https://github.com/medusajs/medusa/pull/2773